### ISSUE TITLE
DAC noise

### ISSUE THREAD

**Issue by hartytp** [sinara-hw/Stabilizer#24]
Breaking the DAC noise I reported in https://github.com/sinara-hw/Stabilizer/issues/9 out into its own issue:

With DAC 0 at +10V I see the following (AC-coupled) output driving a 1M scope:
![dac_0_noise](https://user-images.githubusercontent.com/21218399/60723537-cf30c200-9f2b-11e9-8a19-cf40cd9a1b1c.jpg)

- I see approx 7.5mV of noise at 140kHz
- This is with Stabilizer powered from a 12V supply, but I see the same with PoE
- No difference with ethernet + JTAG connected or disconnected
- No difference when I place a jumper across pins 9 and 10 of the analog header to short DAC_0_N to ground
- The noise I see scales with the DAC output voltage, so this seems like it must be noise on the reference. I'll post shots of different output voltages in a moment
- @gkasprow had a go at reproducing this in #9 but I do not yet trust his measurements since he wasn't getting the correct DC output. @jordens have you seen this noise in any of your tests?

**Comment by hartytp** [comment]
Moved the setup to a different room. The output looks much cleaner (far fewer SMPS spikes visible) but the 140kHz noise unchanged. So I don't think it's environmental/grounding related. The good news is that, other than this noise, the output seems pretty clean...
![Screenshot](https://user-images.githubusercontent.com/21218399/60724354-1455f380-9f2e-11e9-9f34-7b44f4e15b2d.jpg)

**Comment by hartytp** [comment]
I've turned on some digital averaging on my scope to make the noise we're looking at a little clear...

channel 0 and channel 1 both at 0xffff (+10V):
![image](https://user-images.githubusercontent.com/21218399/60725659-94ca2380-9f31-11e9-9974-8b75473a8401.png)


0xBFFF (+5V):

![image](https://user-images.githubusercontent.com/21218399/60725621-7cf29f80-9f31-11e9-874a-e6635f891963.png)

**Comment by hartytp** [comment]
0x7FFF (0V):

http://10.255.6.27/GetImage
![image](https://user-images.githubusercontent.com/21218399/60725856-030ee600-9f32-11e9-9c35-f066657f2172.png)

**Comment by hartytp** [comment]
-10V (0x0000):

![image](https://user-images.githubusercontent.com/21218399/60725902-2174e180-9f32-11e9-8b9d-5e8fee375e8c.png)

**Comment by hartytp** [comment]
So, it does seem that there is some noise on the reference that's getting scaled by the DAC.

NB this was all done with the quartiq/master firmware changing a couple of lines to manually set the DAC.

**Comment by hartytp** [comment]
Indeed, the noise is on the reference. here, the yellow and green traces are the two DAC outputs, with the DAC code set to 0x0000. The blue is a 10x scope probe looking at teh 100nF reference cap by DAC pin 5, the grey is a zoom in of that. Lots of averaging to pull the signal out.

![image](https://user-images.githubusercontent.com/21218399/60727464-d78dfa80-9f35-11e9-823e-61b4fecc9677.png)

**Comment by hartytp** [comment]
@gkasprow oops.... I think this is the chopper-stabilizer amp you're using to servo out the LT3042 offsets. Look at figure 3 on the data sheet. https://www.analog.com/media/en/technical-documentation/data-sheets/ADA4528-1_4528-2.pdf

I'll depopulate some resistors to remove that reference buffer and confirm...

**Comment by hartytp** [comment]
Indeed, that's the issue.

Same, but with R158 removed (inc heavy averaging)...

![image](https://user-images.githubusercontent.com/21218399/60728934-736d3580-9f39-11e9-99bd-b072bdc6fa6b.png)

**Comment by hartytp** [comment]
aah...no...removing R158 stopped the board from powering on correctly, as Rset for the LT3042 is only 39K. Replacing it with 41k allows the board to start and I see the same noise as before. So not that buffer after all...

**Comment by hartytp** [comment]
I think it may be the LT3042 oscillating slightly. May not like such a capacative load.

The voltage on the set pin is clean, but the oscillations are clearly visible on the output.

**Comment by hartytp** [comment]
Noise also visible on C139 (tantalum input capacitor). Maybe insufficient input capacitance. I'll try adding a large ceramic input capacitor in parallel with the 100nF.

**Comment by hartytp** [comment]
Definitely an unhappy ref buffer. Definitely too much or too little capacitance somewhere...

**Comment by gkasprow** [comment]
It didn't oscillate on my board.
Such LDOs tend to oscillate when too low ESR capacitor is used.
You can replace R94, R34, R105, R106 with 0.25R or even 1R and see if that helps
You can also check if IC27 is oscillating

**Comment by gkasprow** [comment]
edit. Didn't read that you already removed the R158. So I'd try to increase these resistors or even remove them to see if that helps.

**Comment by gkasprow** [comment]
5V is generated from SMPS, so check what is the frequency of it. But it should be far more than 140KHz. It is also passed via LDO so I'd be surprised if it was the source. Maybe IC16 is oscillating? Are the oscillations at the ref buffer input higher or lower than on the output?

**Comment by hartytp** [comment]
Here is the P5V0 test point: 
![image](https://user-images.githubusercontent.com/21218399/60731738-39ebf880-9f40-11e9-9461-9ed7ba055fb9.png) 

here is the LT3042 output:

![image](https://user-images.githubusercontent.com/21218399/60731766-4f612280-9f40-11e9-8eb1-aa15e04eee1e.png)

**Comment by hartytp** [comment]
So it seems that the LT3042 has almost no power supply rejection at the frequency. It's possible that the high-frequency PSRR is very low due to the heavy capacitive load.

**Comment by hartytp** [comment]
> It didn't oscillate on my board.

A few mV is easy to miss when probing with a scope. So, it may have been there but hard to spot.

**Comment by gkasprow** [comment]
I was observing it with 1mV/div scale and 20MHz BW.
I'd try with disconnecting the loads first.

**Comment by hartytp** [comment]
hmmm... from the data sheet https://www.analog.com/media/en/technical-documentation/data-sheets/3042fb.pdf the PSRR should still be 80dB at 200kHz with 22uF output capacitance, so this doesn't seem at all right.

**Comment by hartytp** [comment]
![Screenshot from 2019-07-05 16-26-06](https://user-images.githubusercontent.com/21218399/60732339-a5829580-9f41-11e9-9253-72b665babfbb.png)

The PSRR drops off very rapidly with deceasing drop out voltage. However, on my board the 5V0A rail is at 4.975V, the ref output is at 4.3V. So, that's about 700mV head room, so I'd expect around 60dB.

I'll try disconnecting the loads and increasing C140 to 10uF (to ensure stability) next.

**Comment by gkasprow** [comment]
I don't think it is caused by the input signal. The LDO oscillates dragging current from the input.

**Comment by dnadlinger** [comment]
Ah, the minimum output capacitance is 4.7 μF – the regulator loop bandwidth is about 1 MHz, so it's possible the ESR/ESL to the ADC buffer caps is high enough to render the loop unstable with only 1 µF output capacitance.

**Comment by hartytp** [comment]
@gkasprow  Yes. I believe that the output capacitance on the LT3042 is not sufficient and that's causing it to oscillate. It requires a 4.7uF ceramic output capacitor (ESR < 50mOhm and ESL < 2nH). However, we have a 1uF ceramic and then some 10uF tantalums quite far away.

Replacing the 1uF with 10uF the problem goes away.

@gkasprow the reason you didn't see this may have been that the circuit is boarderline stable so capacitor tolerances etc can cause it to move in/out of stability.

**Comment by hartytp** [comment]
@dnadlinger indeed.

**Comment by gkasprow** [comment]
This is this type of issue that appears once you produce more units...

**Comment by hartytp** [comment]
Here is the outputs of channels 0 and 1, both set to 0xffff

![image](https://user-images.githubusercontent.com/21218399/60735231-699ffe00-9f4a-11e9-8fcc-c30457f7d3de.png)

I'll take more data in a quieter environment once i've reverted all my hardware modifications apart from the 10uF output capacitor.

**Comment by hartytp** [comment]
Good news is that the output noise is now 1LSB RMS on both DACs, so it's actually a 16-bit system.

Same but different time bases

**Comment by hartytp** [comment]
outputs look same at 0V as 10V. Good

**Comment by gkasprow** [comment]
can you make FFT?

**Comment by hartytp** [comment]
![image](https://user-images.githubusercontent.com/21218399/60735768-73c2fc00-9f4c-11e9-8fe9-3a03afc69127.png)

here you go.

**Comment by gkasprow** [comment]
can you male a loop from crocodile GND clip and approach it to the SMPS converters and see if the 2 peaks on FFT are caused by them?
I'm also curious what this 8MHz peak comes from. Can you switch off the board and see if it is still there?

**Comment by hartytp** [comment]
This is with heavy averaging on obviously.

There is a small SMPS peak at ~500kHz and a bump at 8MHz.

**Comment by hartytp** [comment]
board off 
![image](https://user-images.githubusercontent.com/21218399/60736213-1465eb80-9f4e-11e9-90eb-1dfc58256a2b.png)

board on
![image](https://user-images.githubusercontent.com/21218399/60736247-352e4100-9f4e-11e9-9876-85aaf74b0f70.png)

**Comment by hartytp** [comment]
closer in noise, board off 
![image](https://user-images.githubusercontent.com/21218399/60736298-5f7ffe80-9f4e-11e9-8060-2c2dacc7816a.png)

board on 
![image](https://user-images.githubusercontent.com/21218399/60736313-6ad32a00-9f4e-11e9-8c87-2852cdf33f63.png)

So there is definitely a bit of 500kHz getting through somewhere. I'll take more data in a quieter room in a bit.

**Comment by gkasprow** [comment]
can you disable CMC?

**Comment by hartytp** [comment]
Shorted CMC:

![image](https://user-images.githubusercontent.com/21218399/60736488-11b7c600-9f4f-11e9-9503-8a968405de0a.png)


unshort:
![image](https://user-images.githubusercontent.com/21218399/60736510-2ac07700-9f4f-11e9-9dbf-d6232e962418.png)

So in this particular case the CMC does make things ever so slightly worse

**Comment by hartytp** [comment]
The 500kHz peak is still the same.

Anyway, there are still some noise peaks there, but pretty close to the noise floor. 

I'd have to check my noise model, but I guess we expect something like 15nv/rtHz output noise due to the 10k resistors, OpAmp etc. 

With stabilizer unplugged, I see a scope noise floor of 160uV RMS in a 20MHz BW (brick wall filter). With stabilizer I see 360uV. Implies stabilizer RMS in 20MHz is 330uV (scope noise is negligible). Assume noise roughly white. Noise floor is 70nV/rtHz. That's high. I'd expect more like 15nV/rtHz based on the OpAmp, resistors etc.
To put it another way

**Comment by hartytp** [comment]
@gkasprow the 500kSPS is the DAC update rate, so that will be a little digital feed through sneaking in just above the noise floor.

**Comment by gkasprow** [comment]
makes sense.

**Comment by hartytp** [comment]
Putting it on a UPV audio analyzer with floating input, I measure 100nV/rtHz white noise (instrument BW is 250kHz). Instrument noise floor is moer like 10nV/rtHz.

Gain of AFE is 5, so that's about 20nV/rtHz in-band noise. That's about right given a 12nV/rtHz for the DAC, 6nV/rtHz for each OpAmp and 12nV/rtHz for each 10k resistor.

100nV/rtHz*sqrt(500kHz) = 70uV rms. So, still a bit lower than I measured. But, might be due to grounding etc. Anyway, I don't see anything particularly worrying here, so I'm happy to call a day on this.

**Comment by hartytp** [comment]
Basically, this all works as well as we can hope...good!

**Comment by gkasprow** [comment]
Great! Next well-engineered module thanks to the collaborative effort.

**Comment by hartytp** [comment]
Note to self, to do at some point next week:
- put LT3042 OpAmp components back
- put LT3042 input capacitors back to original values (10uF tant + 100nF ceramic)
- check noise with PoE (CMC shorted and unshorted)
- replace/remove ADC buffers

**Comment by gkasprow** [comment]
@hartytp what's the conclusion then? Which components should I change at the reference buffer?

**Comment by gkasprow** [comment]
I found it. I should increase LT3042 output capacitance to at least 4.7uF
